cmd/kubeadm/app/cmd/init.go

```diff
@@ -336,8 +339,7 @@ func (i *Init) Run(out io.Writer) error {
 		"KubeConfigName": kubeadmconstants.AdminKubeConfigFileName,
 		"Token":          i.cfg.Token,
 		"CAPubKeyPin":    pubkeypin.Hash(caCert),
-		"MasterIP":       i.cfg.API.AdvertiseAddress,
-		"MasterPort":     strconv.Itoa(int(i.cfg.API.BindPort)),
+		"MasterEndpoint": strings.TrimPrefix(masterEndpoint, "https://"),
```

Comment:
Rather than call GetMasterEndpoint(), which adds https:// prefix, and then strip it out here, I'd suggest creating an inner function that creates the IP/port pair (used here), and an outer function that adds the https:// prefix that is used in other areas.

The advantage is that, with this separation, you can have unit tests that verify that this map is correct. As it stands now, we don't have code coverage on L342, which would ensure that the usage in join commands is correct. With a separate function for this case, we can create UTs to ensure no regressions.